src/conv/problem_description.cpp

```diff
@@ -95,7 +110,16 @@ void ProblemDescription::Serialize(std::ostream& stream) const
     stream << sep << PrintDHW('x', GetSpatialDims(), GetKernelStrideD(), GetKernelStrideH(), GetKernelStrideW());
     stream << sep << PrintDHW('x', GetSpatialDims(), GetDilationD(), GetDilationH(), GetDilationW());
     stream << sep << GetBias();
-    stream << sep << GetInLayout();
+    if (GetInLayout() == "NCHW" && GetWeightsLayout() == "KCYX" && GetOutLayout() == "NKHW")
+    {
+        stream << sep << GetInLayout();
+    }else {
+        stream << sep << GetInLayout();
+        stream << sep << GetWeightsLayout();
+        stream << sep << GetOutLayout();
+    }
+    stream << sep << GetWeightsLayout();
+    stream << sep << GetOutLayout();
```

Comment:
For backwards compatibility the output layout should not be serialized either.

Reply:
Thanks for catching this! Think I accidentally forgot about those two lines when adding the if condition. Fixed in d0f3423